### PR TITLE
fix of issue 1290

### PR DESCRIPTION
**Pull request by bingbing8**
https://github.com/PowerShell/Win32-OpenSSH/issues/1290

### CODE REVIEW

File: contrib/win32/win32compat/win32_pty.c [hunk 1]
```diff
@@ -154,34 +154,38 @@ int exec_command_with_pty(int * pid, char* cmd, int in, int out, int err, unsign
...
-
-	_snwprintf_s(pty_cmdline, MAX_CMD_LEN, MAX_CMD_LEN, L"\"%ls\\ssh-shellhost.exe\" ---pty %ls", __wprogdir, cmd_w);
-	/* 
-	 * In PTY mode, ssh-shellhost takes stderr as control channel
-	 * TODO - fix this and pass control channel pipe as a command line parameter
-	 */
-	si.hStdError = ttyh;
-	debug3("pty commandline: %ls", pty_cmdline);
+	do {
+		if (is_conpty_supported()) {
```
Comment (bagajjal) [inline contrib/win32/win32compat/win32_pty.c]: Have if / else instead of do/while

File: contrib/win32/win32compat/win32_pty.c [hunk 2]
```diff
@@ -154,34 +154,38 @@ int exec_command_with_pty(int * pid, char* cmd, int in, int out, int err, unsign
...
+			ret = CreateConPty(cmd_w, (short)si.dwXCountChars, (short)si.dwYCountChars, si.hStdInput, si.hStdOutput, ttyh, &pi);
```
Comment (bagajjal) [inline contrib/win32/win32compat/win32_pty.c]: In CreateConPty() handles are not closed.. We need to fix that

File: contrib/win32/win32compat/win32_pty.c (hunk 1 above)
Comment (bingbing8) [reply contrib/win32/win32compat/win32_pty.c]: changed to if else

File: contrib/win32/win32compat/win32_pty.c (hunk 2 above)
Comment (bingbing8) [reply contrib/win32/win32compat/win32_pty.c]: I combined the conpty into exec_command_with_pty and close handles in exec_command_with_pty.

File: contrib/win32/win32compat/win32_pty.c [hunk 3]
```diff
@@ -180,10 +145,12 @@ int exec_command_with_pty(int * pid, char* cmd, int in, int out, int err, unsign
 		errno = EOTHER;
 		goto done;
 	}
+
 	*pid = pi.dwProcessId;
 	ret = 0;
 
 done:
+	SetConsoleCtrlHandler(NULL, TRUE);
```
Comment (bingbing8) [inline contrib/win32/win32compat/win32_pty.c:152]: @bagajjal , is this okay to call this on shellhost case? or I need to add a condition to call it only for conpty case?

Comment (bagajjal) [reply contrib/win32/win32compat/win32_pty.c:152]: I think we do this inside shellhost but you can verify it once..
When you are using ssh-shellhost.. 
1) launch ping -t bing.com
2) Now issue CTRL+C
3) If we are able to stop the ping then we are good..

**Review by bagajjal — APPROVED** [review]

File: contrib/win32/win32compat/win32_pty.c (hunk 3 above)
Comment (bingbing8) [reply contrib/win32/win32compat/win32_pty.c:152]: confirm that ssh-shellhost has it and CTRL+C works fine.

**Pull request merged**